tests/handlers/records-subscribe.spec.ts

```diff
@@ -0,0 +1,773 @@
+          // scenario: Alice writes some chat messages and gives Bob a role allowing him to access them. But Bob's filter
+          //           does not contain a contextId so the subscription fails.
+          const alice = await DidKeyResolver.generate();
+          const bob = await DidKeyResolver.generate();
+
+          const protocolDefinition = threadRoleProtocolDefinition;
+
+          const protocolsConfig = await TestDataGenerator.generateProtocolsConfigure({
+            author: alice,
+            protocolDefinition
+          });
+          const protocolsConfigureReply = await dwn.processMessage(alice.did, protocolsConfig.message);
+          expect(protocolsConfigureReply.status.code).to.equal(202);
+
+          // Alice writes a 'thread' record
+          const threadRecord = await TestDataGenerator.generateRecordsWrite({
+            author       : alice,
+            protocol     : protocolDefinition.protocol,
+            protocolPath : 'thread',
+          });
+          const threadRoleReply = await dwn.processMessage(alice.did, threadRecord.message, { dataStream: threadRecord.dataStream });
+          expect(threadRoleReply.status.code).to.equal(202);
+
+          // Alice writes a 'friend' $globalRole record with Bob as recipient
+          const participantRoleRecord = await TestDataGenerator.generateRecordsWrite({
+            author       : alice,
+            recipient    : bob.did,
+            protocol     : protocolDefinition.protocol,
+            protocolPath : 'thread/participant',
+            contextId    : threadRecord.message.contextId,
+            parentId     : threadRecord.message.recordId,
+            data         : new TextEncoder().encode('Bob is my friend'),
+          });
+          const participantRoleReply =
+            await dwn.processMessage(alice.did, participantRoleRecord.message, { dataStream: participantRoleRecord.dataStream });
+          expect(participantRoleReply.status.code).to.equal(202);
+
+          // Alice writes three 'chat' records
+          const chatRecordIds = [];
+          for (let i = 0; i < 3; i++) {
+            const chatRecord = await TestDataGenerator.generateRecordsWrite({
+              author       : alice,
+              recipient    : alice.did,
+              protocol     : protocolDefinition.protocol,
+              protocolPath : 'thread/chat',
+              published    : false,
+              contextId    : threadRecord.message.contextId,
+              parentId     : threadRecord.message.recordId,
+              data         : new TextEncoder().encode('Bob can read this cuz he is my friend'),
+            });
+            const chatReply = await dwn.processMessage(alice.did, chatRecord.message, { dataStream: chatRecord.dataStream });
+            expect(chatReply.status.code).to.equal(202);
+            chatRecordIds.push(chatRecord.message.recordId);
+          }
```

Comment:
Consider remove.

Reply:
Makes sense, done!

Reply:
There is one more block of this (starting line 563).

Reply:
Woops! Done!